e2e/test/scenarios/filters-reproductions/dashboard-filters-with-question-revert.cy.spec.js

```diff
@@ -0,0 +1,323 @@
+          setFilter("Number");
+          connectFilterToColumn("Field-mapped Rating");
+          saveDashboard();
+
+          cy.log("Give it a value and make sure that the filer applies");
+          filterWidget().click();
+          popover().within(() => {
+            cy.findByPlaceholderText("Enter a number").type(3).blur();
+            cy.button("Add filter").click();
+          });
+          assertFilterIsApplied();
+
+          cy.log("Drill down to the question from the dashboard");
+          cy.findByTestId("legend-caption-title").click();
+          cy.get("@questionId").then(id => {
+            cy.location("pathname").should(
+              "eq",
+              `/question/${id}-${questionDetails.name}`,
+            );
+            cy.location("search").should("eq", "?RATING=3");
+          });
+
+          cy.log("Revert the question to its original (GUI) version");
+          cy.intercept("POST", "/api/revision/revert").as("revertQuestion");
+          questionInfoButton().click();
+          cy.findByTestId("saved-question-history-list")
+            .find("li")
+            .filter(":contains(You created this)")
+            .findByTestId("question-revert-button")
+            .click();
+          cy.wait("@revertQuestion");
+          // Mid-test assertions to root out the flakiness
+          cy.findByTestId("saved-question-history-list").should(
+            "contain",
+            "You edited this",
+          );
+          cy.findByTestId("saved-question-history-list")
+            .findAllByTestId("question-revert-button")
+            .should("have.length", 2);
+
+          cy.findByLabelText(`Back to ${dashboardDetails.name}`).click();
+
+          cy.get("@dashboardId").then(id => {
+            cy.location("pathname").should(
+              "eq",
+              `/dashboard/${id}-${dashboardDetails.name.toLowerCase()}`,
+            );
+            cy.location("search").should("eq", "?equal_to=3");
+          });
+
+          cy.log("Make sure the disconnected filter doesn't break UI");
+          // The filter widget will still have the number 3 applied as the filter,
+          // but that shouldn't affect our results since the filter is disconnected.
+          assertFilterIsDisconnected();
+
+          // Reloading the dashboard breaks the filter in the original issue
+          cy.reload();
+
+          cy.log(
+            "Make sure the UI shows the filter is not connected to the GUI card",
+          );
+          editDashboard();
+
+          cy.findByTestId("fixed-width-filters").icon("gear").click();
+          getDashboardCard().should("contain", "Unknown Field");
+
+          snapshot("35954");
+        });
+      });
+
+      beforeEach(() => {
+        restore("35954");
+        cy.signInAsAdmin();
+      });
+
+      it("should be able to remove the broken connection and connect the filter to the GUI question", function () {
+        visitDashboard(this.dashboardId);
```

Comment:
Using JS functions gives us access to the previously `wrap`-ed values from the `before` block.